objectherkenning_openbare_ruimte/performance_evaluation_pipeline/components/evaluate_model.py

```diff
@@ -0,0 +1,120 @@
+    per_image_result_to_df,
+    tba_result_to_df,
+)
+from objectherkenning_openbare_ruimte.settings.settings import (  # noqa: E402
+    ObjectherkenningOpenbareRuimteSettings,
+)
+
+config_path = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), "..", "..", "..", "config.yml")
+)
+
+ObjectherkenningOpenbareRuimteSettings.set_from_yaml(config_path)
+settings = ObjectherkenningOpenbareRuimteSettings.get_settings()
+
+log_settings = settings["logging"]
+azure_logging_configurer = AzureLoggingConfigurer(settings["logging"])
+azure_logging_configurer.setup_oor_logging()
+logger = logging.getLogger("performance_evaluation")
+
+aml_experiment_settings = settings["aml_experiment_details"]
+
+
+@command_component(
+    name="evaluate_model",
+    display_name="Evaluate model predictions.",
+    environment=f"azureml:{aml_experiment_settings['env_name']}:{aml_experiment_settings['env_version']}",
+    code="../../../",
+    is_deterministic=True,
+)
+def evaluate_model(
+    ground_truth_base_dir: Input(type=AssetTypes.URI_FOLDER),  # type: ignore # noqa: F821
+    predictions_base_dir: Input(type=AssetTypes.URI_FOLDER),  # type: ignore # noqa: F821
+    output_dir: Output(type=AssetTypes.URI_FOLDER),  # type: ignore # noqa: F821
+):
+    """
+    Run evaluation of a model based on ground truth annotations and model
+    predictions.
+
+    This pipeline runs three evaluation methods:
+
+    * Total Blurred Area evaluation for sensitive classes. This tells us the
+      percentage of bounding boxes that are covered by predictions.
+
+    * Per Image evaluation. This tells us the precision and recall based on
+      whole images, i.e. if a single image contains at least one annotation of a
+      certain class, does it also contain at least one prediction.
+
+    * Custom COCO evaluation. This is a COCO-style evaluation of overall and per
+      class precision and recall, for different bounding box sizes and
+      confidence thresholds.
+
+    Results are stored as CSV files in the chosen output location.
+
+    Parameters
+    ----------
+
+    ground_truth_base_dir: Input(type=AssetTypes.URI_FOLDER)
+        Location of ground truth dataset (root folder, is expected to contain
+        `images/` and `labels/` subfolders).
+    predictions_base_dir: Input(type=AssetTypes.URI_FOLDER)
+        Location of predictions (root folder, is expected to contain `labels/`
+        subfolder).
+    output_dir: Output(type=AssetTypes.URI_FOLDER)
+        Location where output will be stored.
+    """
+    model_name = settings["performance_evaluation"]["model_name"]
+    predictions_img_shape = settings["performance_evaluation"][
+        "predictions_image_shape"
+    ]
+    prediction_labels_rel_path = settings["performance_evaluation"][
+        "prediction_labels_rel_path"
+    ]
+
+    logger.info(f"Running performance evaluation for model: {model_name}")
+
+    os.makedirs(output_dir, exist_ok=True)
+
+    OOREval = OOREvaluation(
```

Comment:
Nice stuff! I would only suggest here some name changes:
- the class would be maybe better OOREvaluator, because it's the object that performs the evaluation;
- the instantiated object by convention should be lower case and snake, so oor_eval;
- the functions of a class normally I like name them active, like evaluate_tba instead of tba_evaluation.

Just details (except the second point) and if you don't like it don't change it 😉

Reply:
Good points :)